### PR TITLE
document DMD style more prominently

### PR DESCRIPTION
**Pull request by thewilsonator**
#9511 minus the controversial items.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,6 +66,106 @@ For first-time contributers, we suggest to look for issues categorized as [trivi
...
+2. Use correct Ddoc function comment blocks. Do not use Ddoc comments for
+overrides unless the overriding function does something different (as far as
+the caller is concerned) than the overridden function. Ddoc comment blocks
+are often overkill for nested functions and function literals; use ordinary
+comments for those. Follow the [D Style](https://dlang.org/dstyle.html#phobos_documentation)
+for for comment blocks. 
+
+3. Do not use `strlen`/`strcmp` and their ilk. Use D arrays instead.
+
+4. Use `ref` instead of raw pointers.
```
Comment (jacob-carlborg) [inline CONTRIBUTING.md]: Add`out` as well.

**Review by wilzbach — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (thewilsonator) [reply CONTRIBUTING.md]: Done.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by jacob-carlborg** [comment]
Half of the tests don't pass 😢.

**Comment by thewilsonator** [comment]
DAutoTest has a network problem, circle fails regularly for network and environmental reasons, semaphore fails because bootstrap with GDC8 got broken recently.

**Pull request merged**